graphql/resolve_field.go

```diff
@@ -28,8 +28,15 @@ func ResolveField[T any](
 		recoverFromPanic,
 		nonNull,
 		Null,
-		func(ctx context.Context, res T) Marshaler {
-			return marshal(ctx, field.Selections, res)
+		func(ctx context.Context, res func() (T, error)) Marshaler {
+			return WriterFunc(func(w io.Writer) {
+				ret, err := res()
```

Comment:
This circumvents the non null check that errors with "must not be null" in `resolveField` because this function could return nil